### PR TITLE
GCP: Storage Bucket Acccess Control

### PR DESCRIPTION
**Pull request by rambleraptor**
##### SUMMARY
Adding support for GCP Storage Bucket Access Controls

(These are ACLs on Storage Buckets)

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
gcp_storage_bucket_access_control

##### ANSIBLE VERSION
```
2.6
```

##### ADDITIONAL INFORMATION
This was autogenerated using [Magic Modules](https://github.com/GoogleCloudPlatform/magic-modules)

### CODE REVIEW

File: lib/ansible/modules/cloud/google/gcp_storage_bucket_access_control.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    module.exit_json(**fetch)
+
+
+def create(module, link, kind):
+    auth = GcpSession(module, 'storage')
+    return return_if_object(module, auth.post(link, resource_to_request(module)), kind)
+
+
+def update(module, link, kind):
+    module.fail_json(msg="BucketAccessControl cannot be edited")
```
Comment (ryansb) [inline lib/ansible/modules/cloud/google/gcp_storage_bucket_access_control.py]: How are users supposed to change permissions?

**Review by ryansb — CHANGES_REQUESTED** [review]

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/modules/cloud/google/gcp_storage_bucket_access_control.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: gcp_storage_bucket_access_control
+description:
+    - The BucketAccessControls resource represents the Access Control Lists
+      (ACLs) for buckets within Google Cloud Storage. ACLs let you specify who
+      has access to your data and to what extent.
+    - |
```
Comment (ryansb) [inline lib/ansible/modules/cloud/google/gcp_storage_bucket_access_control.py]: This weird paragraph break is still showing up, can you try regenerating this?

File: lib/ansible/modules/cloud/google/gcp_storage_bucket_access_control.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        required: true
+        choices: ['present', 'absent']
+        default: 'present'
+    bucket:
+        description:
+            - A reference to Bucket resource.
+        required: true
+    entity:
+        description:
+            - |
```
Comment (ryansb) [inline lib/ansible/modules/cloud/google/gcp_storage_bucket_access_control.py]: Same

**Comment by rambleraptor** [comment]
Fixed doc issues on this PR (and others)!

**Review by ryansb — APPROVED** [review]

**Pull request merged**